tests/integration-tests/tests/configure/test_pcluster_configure.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+import logging
+
+import pexpect
+import pytest
+from assertpy import assert_that
+
+
+@pytest.mark.regions(["us-east-1"])
+@pytest.mark.instances(["c5.xlarge", "m6g.xlarge"])
+@pytest.mark.schedulers(["awsbatch", "slurm"])
+def test_pcluster_configure(vpc_stack, key_name, region, os, instance, scheduler, clusters_factory, test_datadir):
+    """Verify that the configuration file produced by `pcluster configure` can be used to create a cluster."""
+    config_path = test_datadir / "config.ini"
+    orchestrate_pcluster_configure(
+        region,
+        key_name,
+        scheduler,
+        os,
+        instance,
+        vpc_stack.cfn_outputs["VpcId"],
+        vpc_stack.cfn_outputs["PublicSubnetId"],
+        vpc_stack.cfn_outputs["PrivateSubnetId"],
+        config_path,
+    )
+    clusters_factory(config_path)
```

Comment:
shall we also compare the content of the config file against the expected config other than creating the cluster? This would have caught the issue with the metadata config option.

Reply:
I will add a check to ensure that the subset of parameters explicitly configured by the test have the expected values. As for checking for unexpected parameters: the unit tests were updated to ensure only the expected configs are being written to the file. That is sufficient in my opinion.